### ISSUE TITLE
chore: remove GPL2-only CAISI note permission scoping system (NotePermission cluster)

### ISSUE THREAD

**Issue by yingbull** [carlos-emr/carlos#558]
## Summary

Remove the GPL2-only (Indivica 2008-2012) CAISI note permission scoping system — a cluster of 4 files plus inline references across 5 more files. This feature allows per-note program/role assignment in multi-program CAISI deployments. It is gated behind the `note_program_ui_enabled` property (defaults to `true`) and is unnecessary for standard circle-of-care clinic deployments.

**License motivation**: All 4 primary files are **GPL v2 only** (not GPL2+), blocking relicensing. The core note creation, viewing, and read-enforcement code is entirely GPL2+ (McMaster/CAISI).

**Functional impact**: Removes the UI for viewing/changing which CAISI program a note belongs to, and the "View Note Scope" permission summary popup. Notes will still be auto-assigned to a program via existing GPL2+ code paths. Does NOT affect note reading, writing, signing, or the standard security model.

---

## Files to Delete (4 GPL2-only files)

| File | License | Purpose |
|------|---------|---------|
| `src/main/java/io/github/carlos_emr/carlos/casemgmt/web/NotePermission.java` | GPL2 (Indivica) | Enum model class (AccessType: ALL_ROLES, ROLE, ROLE_GLOBAL, NO_ACCESS) |
| `src/main/java/io/github/carlos_emr/carlos/casemgmt/web/NotePermissions2Action.java` | GPL2 (Indivica) | Struts2 action: getNoteScope, setNotePermissions, getDefaultProgramAndRole, visibleProgramsAndRoles |
| `src/main/webapp/casemgmt/noteProgram.js` | GPL2 (Indivica) | Client-side AJAX calls to NotePermissions.do, popup UI logic |
| `src/main/webapp/casemgmt/noteProgram.css` | GPL2 (Indivica) | Styles for `._program` label, popup, and scope overlay |

### 1. `src/main/java/io/github/carlos_emr/carlos/casemgmt/web/CaseManagementEntry2Action.java` (GPL2+, CAISI)

**Two changes needed:**

**A. Replace static method call with inline copy (lines 768, 3219)**

Lines 768 and 3219 both call:
```java
NotePermissions2Action.getAllProviderAccessibleRolesForDemo(providerNo, demographicNo);
```

This static method resolves which programs/roles a provider can access for a given patient. `NotesService.java` (GPL2+) already has an **independent private copy** of this exact method at line 1773 — it does NOT import from NotePermissions2Action. 

**Action**: Copy `getAllProviderAccessibleRolesForDemo()` (~50 lines) from `NotesService.java:1773-1824` into `CaseManagementEntry2Action` as a private method. Update both call sites (768, 3219) to use the local copy. Remove the import of `NotePermissions2Action`.

**B. Remove the `note_program_ui_enabled` parameter override block (lines 1440-1455)**

This block reads `_note_program_no` and `_note_role_id` from the form submission — parameters that are only populated by the `noteProgram.js` UI being removed. Without the JS populating these hidden fields, this block would throw an exception on every note save.

```java
// REMOVE this entire block (lines 1440-1455):
if (OscarProperties.getInstance().getBooleanProperty("note_program_ui_enabled", "true")) {
    String noteProgramNo = request.getParameter("_note_program_no");
    String noteRoleId = request.getParameter("_note_role_id");
    // ... validation and assignment ...
}
```

The auto-assignment logic at lines 762-787 (and the fallback at 3200-3233) already handles program/role assignment via `getAllProviderAccessibleRolesForDemo()` — the removed block was an *override* that let the UI trump the auto-assignment. Without the UI, the auto-assignment is sufficient.

### 2. `src/main/webapp/casemgmt/newEncounterLayout.jsp` (GPL2+, CAISI)

**Two blocks to remove:**

**A. Script/CSS includes (lines 170-173):**
```jsp
<% if (OscarProperties.getInstance().getBooleanProperty("note_program_ui_enabled", "true")) { %>
<link rel="stylesheet" href="<c:out value="${ctx}/casemgmt/noteProgram.css" />"/>
<script type="text/javascript" src="<c:out value="${ctx}/casemgmt/noteProgram.js" />"></script>
<% } %>
```

**B. Popup and scope overlay HTML (lines 875-918):**
```jsp
<%
    if (OscarProperties.getInstance().getBooleanProperty("note_program_ui_enabled", "true")) {
%>
<span class="popup" style="display: none;" id="_program_popup">
    ... program/role select dropdown popup ...
</span>
<div id="_program_scope" class="_program_screen" style="display: none;">
    ... "Note Permission Summary" overlay ...
</div>
<%
    }
%>
```

### 3. `src/main/webapp/casemgmt/ChartNotesAjax.jsp` (GPL2+, McMaster)

**Five `note_program_ui_enabled` gated blocks to remove:**

**A. `_setupNewNote()` call for new notes (lines 398-403):**
```jsp
<% if (OscarProperties.getInstance().getBooleanProperty("note_program_ui_enabled", "true")) { %>
<script>_setupNewNote();</script>
<% } %>
```

**B. Per-note `._program` div rendering (lines 713-720):**
```jsp
<% if (OscarProperties.getInstance().getBooleanProperty("note_program_ui_enabled", "true")) { %>
<div class="_program" noteId="<%=globalNoteId %>" programName="<%=note.getProgramName() %>"
     roleName="<%=note.getRoleName() %>">
    <span class="program"><%=note.getProgramName() %> (<%=note.getRoleName() %>)</span>
</div>
<% } %>
```

**C. `_setupNewNote()` call at end of note block (lines 888-892):**
```jsp
<% if (OscarProperties.getInstance().getBooleanProperty("note_program_ui_enabled", "true")) { %>
<script>_setupNewNote();</script>
<% } %>
```

**D. `_setupProgramList()` call (lines 1062-1065):**
```jsp
<% if (OscarProperties.getInstance().getBooleanProperty("note_program_ui_enabled", "true")) { %>
    _setupProgramList();
<% } %>
```

**E. Click handler rebinding (lines 1070-1075):**
```jsp
<% if (OscarProperties.getInstance().getBooleanProperty("note_program_ui_enabled", "true")) { %>
<script type="text/javascript">
    jQuery("._program .program").unbind("click");
    jQuery("._program .program").click(_noteProgramClick);
</script>
<% } %>
```

### 4. `src/main/webapp/casemgmt/ChartNotes.jsp` (GPL2+, McMaster)

**Remove hidden form fields (lines 593-598):**
```jsp
<%
    if (OscarProperties.getInstance().getBooleanProperty("note_program_ui_enabled", "true")) {
%>
<input type="hidden" name="_note_program_no" value=""/>
<input type="hidden" name="_note_role_id" value=""/>
<% } %>
```

### 5. `src/main/webapp/js/newCaseManagementView.js.jsp` (GPL2+, McMaster)

**Three blocks to remove/simplify:**

**A. `_setCurrentProgramAndRoleIdForNote` call (lines 1932-1935):**
```javascript
// Only works with "note_program_ui_enabled = true" (noteProgram.js)
if (typeof _setCurrentProgramAndRoleIdForNote == "function") {
    _setCurrentProgramAndRoleIdForNote(nId);
}
```
This is a safe typeof guard — it already no-ops when the function doesn't exist. **Remove these 4 lines** for cleanliness.

**B. Save validation block #1 (lines 1364-1378):**
```javascript
if ((typeof jQuery("form[name='caseManagementEntryForm'] input[name='_note_program_no']").val() != "undefined") &&
    (typeof jQuery("form[name='caseManagementEntryForm'] input[name='_note_role_id']").val() != "undefined")) {
    if (...) { _missingRoleProgramIdError(); return false; }
    else if (...) { _noVisibleProgramsError(); return false; }
}
```
This checks the hidden fields that no longer exist. The `typeof != "undefined"` guard means it already safely no-ops, but **remove the entire block** since the hidden fields and the JS functions it calls (`_missingRoleProgramIdError`, `_noVisibleProgramsError`) will no longer exist.

**C. Save validation block #2 (lines 3556-3570):**
Same pattern as block B — identical validation logic in a second save path. **Remove the entire block.**

### 6. `src/main/webapp/WEB-INF/classes/struts.xml`

**Remove the action mapping (line 2073):**
```xml
<action name="casemgmt/NotePermissions" class="io.github.carlos_emr.carlos.casemgmt.web.NotePermissions2Action"/>
```

---

## NOT Modified (confirmed independent)

| File | Why It's Unaffected |
|------|-------------------|
| `NotesService.java` (GPL2+) | Has its own private `getAllProviderAccessibleRolesForDemo()` at line 1773. Does NOT import NotePermissions2Action. Its `note_program_ui_enabled` block (line 544) is inside a `/* Currently not available */` comment — already dead code. |
| `CaseManagementView2Action.java` (GPL2+) | Note *read* enforcement. No reference to NotePermission classes. |
| `CaseManagementNote.java` (GPL2+) | Still has `program_no` and `reporter_caisi_role` fields — these remain populated by the auto-assignment logic that stays. |

---

## What Functionality Is Lost

| Capability | Status After Removal |
|-----------|---------------------|
| **Per-note "Program (Role)" label** on notes in the encounter chart | Removed — notes no longer visually show their program/role assignment |
| **Click-to-change program/role** popup for reassigning notes | Removed |
| **"View Note Scope"** permission summary overlay (who can see this note) | Removed |
| **Auto-assignment of program/role** when saving notes | **Preserved** — `CaseManagementEntry2Action` lines 762-787 and 3200-3233 still auto-assign via `getAllProviderAccessibleRolesForDemo()` |
| **Note reading/writing/signing** | **Unchanged** |
| **Note read enforcement** by program | **Unchanged** — handled in `CaseManagementView2Action` (GPL2+) |
| **REST API note creation** | **Unchanged** — `NotesService.java` is fully independent |

**Impact**: Standard circle-of-care clinics see zero change. Multi-program CAISI deployments lose the ability to manually reassign notes between programs via the UI, but auto-assignment and read enforcement remain intact.

---

## Implementation Steps

1. Copy `getAllProviderAccessibleRolesForDemo()` from `NotesService.java:1773-1824` into `CaseManagementEntry2Action` as a private method
2. Update the two call sites in `CaseManagementEntry2Action` (lines 768, 3219) to use the local method; remove the `NotePermissions2Action` import
3. Remove the `note_program_ui_enabled` parameter override block in `CaseManagementEntry2Action` (lines 1440-1455)
4. Remove 5 gated blocks from `ChartNotesAjax.jsp`
5. Remove hidden fields from `ChartNotes.jsp`
6. Remove CSS/JS includes and popup HTML from `newEncounterLayout.jsp`
7. Remove 3 blocks from `newCaseManagementView.js.jsp`
8. Remove struts mapping from `struts.xml`
9. Delete 4 GPL2-only files: `NotePermission.java`, `NotePermissions2Action.java`, `noteProgram.js`, `noteProgram.css`
10. Build and verify: `make install`

## Testing Plan

- [ ] Open an encounter chart — notes display without program/role labels (no JS errors in console)
- [ ] Create a new note — saves successfully, auto-assigned to correct program
- [ ] Edit and sign an existing note — works normally
- [ ] Verify no 404s in browser network tab for `noteProgram.js` or `noteProgram.css`
- [ ] Verify no AJAX calls to `NotePermissions.do` in browser network tab
- [ ] Grep codebase for remaining references: `NotePermission`, `noteProgram`, `_program_popup`, `_program_scope`, `_note_program_no`, `_note_role_id`
- [ ] `make install --run-tests` passes

## Property Cleanup (optional follow-up)

The `note_program_ui_enabled` property can be removed from documentation and any properties templates. After this change, the property is no longer read anywhere except the dead-code comment in `NotesService.java`.

**Comment by yingbull** [comment]
@claude do this work on new branch chore/indivipermsdelete and then pr to develop when done
